[JSC] Align RegExp V Flags Syntax errors with V8

msaboff opened the pull request:
#### 825b9eb20878e31d5064ef2961973ab432af3055
<pre>
[JSC] Align RegExp V Flags Syntax errors with V8
<a href="https://bugs.webkit.org/show_bug.cgi?id=254313">https://bugs.webkit.org/show_bug.cgi?id=254313</a>
rdar://107116409

Reviewed by Yusuke Suzuki.

Updated syntax error tests after comparing to the types of errors that V8 throws.
The changes in errors are either from &quot;invalid class set character&quot; to &quot;invalid operation in class set&quot;
or the opposite, from &quot;invalid operation in class set&quot; to &quot;invalid class set character&quot;.

Found and fixed one case where we weren&apos;t throwing an &quot;invalid class set character&quot; syntax error when
three dashes appear in a row, e.g. /a---x/v.

* JSTests/stress/regexp-vflag-property-of-strings.js:
* Source/JavaScriptCore/yarr/YarrParser.h:
(JSC::Yarr::Parser::consumeAndCheckIfValidClassSetCharacter):
(JSC::Yarr::Parser::parseClassSet):

Canonical link: <a href="https://commits.webkit.org/262017@main">https://commits.webkit.org/262017@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e411fbba821875d6fd8b7319aa1cef43ef5c77e0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/303 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/319 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/332 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/309 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/287 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/314 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/339 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/538 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/309 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/295 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/334 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/283 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/288 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/362 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/282 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/269 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/304 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/316 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/292 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/273 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/322 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/262 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/297 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/67 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/295 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/320 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/39 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (approved):
r=me

rkirsling reviewed (commented):
> The changes in errors are either from "invalid operation in class set" to "invalid operation in class set"
or the opposite, from "invalid operation in class set" to "invalid operation in class set".

Looks like this is meant to be _from "invalid operation in class set" to "invalid class set character"
or the opposite_.

msaboff committed:
[JSC] Align RegExp V Flags Syntax errors with V8
https://bugs.webkit.org/show_bug.cgi?id=254313
rdar://107116409

Reviewed by Yusuke Suzuki.

Updated syntax error tests after comparing to the types of errors that V8 throws.
The changes in errors are either from "invalid class set character" to "invalid operation in class set"
or the opposite, from "invalid operation in class set" to "invalid class set character".

Found and fixed one case where we weren't throwing an "invalid class set character" syntax error when
three dashes appear in a row, e.g. /a---x/v.

* JSTests/stress/regexp-vflag-property-of-strings.js:
* Source/JavaScriptCore/yarr/YarrParser.h:
(JSC::Yarr::Parser::consumeAndCheckIfValidClassSetCharacter):
(JSC::Yarr::Parser::parseClassSet):

Canonical link: https://commits.webkit.org/262017@main
